feat: adds the defaults readme and updates other readmes (#39)
ZupIT/beagle-helpers: android/beagle-defaults/README.md

> # Beagle Scaffold Android

Reviewer: Change the title to Beagle Defaults Defaults

<hr>

> ![Maven Central](https://img.shields.io/maven-central/v/br.com.zup.beagle/beagle-defaults?color=green&label=Beagle-Defaults)

Reviewer 2: ```suggestion
![**Maven Central**](https://img.shields.io/maven-central/v/br.com.zup.beagle/beagle-defaults?color=green&label=Beagle-Defaults)
```

> ![Maven Central](https://img.shields.io/maven-central/v/br.com.zup.beagle/android?label=Beagle)

Reviewer 2: ```suggestion
![**Maven Central**](https://img.shields.io/maven-central/v/br.com.zup.beagle/android?label=Beagle)
```

Here you will find the **Beagle-Defaults** library to help you start a project using Beagle in Android.

> This lib will hold just a few classes necessary 
> to configure and add Beagle into an Android project.

Reviewer 2: ```suggestion
This lib will hold just a few classes necessary to configure and add Beagle into an Android project.
```

### Requirements

> Before you start to configure Beagle for your Android system, it’s important to check out if you have installed all the current versions of the following programs: ‌

Reviewer 2: ```suggestion
Before you start configuring Beagle for your Android system, it’s important to check if you have installed all the current versions of the following dependencies: ‌
```

* JDK 8+ language
* Android API level: at least 19 or higher
* Kotlin 1.3+

### Getting started
<hr>

This tutorial will configure Beagle from the Start.

> Step 1: Add a configuration on the Gradle file in your project:

Reviewer 2: ```suggestion
Step 1: Add a configuration to the Gradle file in your project:
```

- `Add mavenCentral()` into the `All projects` block on your  Project Gradle file(build.gradle)<br>

```kotlin
allprojects {
    repositories {
        google()
        jcenter()
        mavenCentral()
    }
}
```
Step 2: Add kotlin kapt plugin and 3 dependencies on your application dependency manager (gradle.gradle)<br>

>  ![Maven Central](https://img.shields.io/maven-central/v/br.com.zup.beagle/beagle-scaffold?label=beagle-scaffod)

Reviewer 2: ```suggestion
 ![**Maven Central**](https://img.shields.io/maven-central/v/br.com.zup.beagle/beagle-scaffold?label=beagle-scaffod)
```

>  ![Maven Central](https://img.shields.io/maven-central/v/br.com.zup.beagle/android?label=Beagle)                                                                   

Reviewer 2: ```suggestion
 ![**Maven Central**](https://img.shields.io/maven-central/v/br.com.zup.beagle/android?label=Beagle)                                                                   
```

* _The version numbers listed below may have been updated. Please check the actual versions on the tags above:_
```kotlin
apply plugin: 'kotlin-kapt'

android {
    kotlinOptions {
        jvmTarget = JavaVersion.VERSION_1_8
    }
}

dependencies {
    implementation "br.com.zup.beagle:android:1.5.1"
    kapt "br.com.zup.beagle:android-processor:1.5.1"
    implementation "br.com.zup.beagle:beagle-scaffold:0.0.4"
}
```

Step 3: Create 4 configurations files for Beagle.
* These files will set a few configurations and classes used by Beagle.<br>
    * if you need a more detailed information check our documentation

a) Create an AppBeagleConfig file:
```
@BeagleComponent
class AppBeagleConfig: BeagleConfig {
    override val environment: Environment = Environment.DEBUG
    override val baseUrl: String = "https://adopt-beagle.continuousplatform.com/scaffold"
    override val isLoggingEnabled: Boolean = true
    override val cache: Cache = Cache(
        enabled = false,
        maxAge = 300,
        size = 15
    )
}
```

> b) Create an HttpApp file as listed below. This class extends from an ´HttpClientDefault()´ class at the Beagle-Default lib:

Reviewer 3: ```suggestion
b) Create a HttpApp file as listed below. This class extends from a ´HttpClientDefault()´ class at the Beagle-Default lib:
```

Reviewer 2: I believe the use of an instead of a here is correct.

```
@BeagleComponent
class HttpApp: HttpClientDefault()
```

> c) Create a CacheApp file as listed below. This class extends from an ´StoreHandlerDefault()´ class at the Beagle-Default lib:

Reviewer 3: ```suggestion
c) Create a CacheApp file as listed below. This class extends from a ´StoreHandlerDefault()´ class at the Beagle-Default lib:
```

```
@BeagleComponent
class CacheApp : StoreHandlerDefault(AppApplication.APPLICATION!!)
```

> d) Create a LoggerApp file as listed below. This class extends from an ´BeagleLoggerDefault()´ class at the Beagle-Default lib:

Reviewer 3: ```suggestion
d) Create a LoggerApp file as listed below. This class extends from a ´BeagleLoggerDefault()´ class at the Beagle-Default lib:
```
